### DISCUSSION TITLE
List of open source apps using Detox

### DISCUSSION THREAD

**Discussion by ggilchrist-ledger** [wix/Detox#3268] [Q&A]
Hi all, 

Another question from me :) Is there a page or a list of open source apps that use Detox for E2E mobile testing? It would help a lot to see how others have implemented Detox and it's often something that other frameworks provide for helping onboard newbies.

Thanks, Gregor

**Comment by portyu9** [comment]
Hi @ggilchrist-ledger 

A useful production-scale repository to study is **Mattermost Mobile**, which has a dedicated Detox E2E setup and separate Android/iOS E2E workflows.

For smaller examples, also look at Detox's own `examples` directory.

I don't believe Detox currently maintains a canonical, continuously updated showcase page of applications using Detox, so GitHub code search is often the best way to find active projects.

Useful searches include:

```text
detox.config
.detoxrc
detox test
device.launchApp
```

I'd use both kinds of examples:

```text
Detox examples
    -> learn the canonical setup and APIs

large production React Native repos
    -> study helpers, CI integration and test organization
```

That is usually more useful than older "apps using Detox" lists, which tend to become stale.

Please mark this answer as accepted if it helped, thank you.
